include/veriblock/storage/block_repository.hpp

```diff
@@ -0,0 +1,164 @@
+   * operations. It does, when \p BlockRepository::commit is executed on this
+   * batch.
+   */
+  struct WriteBatch {
+    virtual ~WriteBatch() = default;
+
+    /**
+     * Write a single block. If block with such hash exists, db will overwrite
+     * it.
+     * @param block to be written in a batch
+     * @return true if we overwrite existing block, false otherwise.
+     */
+    virtual bool put(const stored_block_t& block) = 0;
+
+    /**
+     * Remove a single block from storage identified by its hash.
+     * @param hash block hash
+     * @return true if removed, false if no such element found.
+     */
+    virtual bool removeByHash(const hash_t& hash) = 0;
+
+    /**
+     * Remove potentially many blocks at given height.
+     * @param height block height
+     * @return true if at least one block removed, false otherwise.
+     */
+    virtual bool removeByHeight(height_t height) = 0;
+
+    /**
+     * Clear batch from any modifying operations.
+     */
+    virtual void clear() = 0;
+  };
+
+  /**
+   * Load a block from disk in memory, by its hash.
+   * @param hash[in] block hash
+   * @param out[out] if non-null, block data will be written here. If null
+   * passed, out argument is ignored.
+   * @return true if block found, false otherwise.
+   */
+  virtual bool getByHash(const hash_t& hash, stored_block_t* out) = 0;
+
+  /**
+   * Load potentially many blocks from disk in memory by their height.
+   * @param height[in] block height
+   * @param out[out] if non-null, blocks will be written here. If null passed,
+   * out argument is ignored.
+   * @return true if block found at given height, false otherwise.
+   */
+  virtual bool getByHeight(const height_t& height,
+                           std::vector<stored_block_t>* out) = 0;
+
+  /**
+   * Load many blocks from disk in memory by a list of hashes.
+   * @param hashes[in] a list of hashes to load.
+   * @param out[out] if non-null, blocks will be written here. If null passed,
+   * out argument is ignored.
+   * @return true if at least one block has been found, false otherwise.
+   */
+  virtual bool getManyByHash(Slice<const hash_t> hashes,
+                             std::vector<stored_block_t>* out) = 0;
+
+  /**
+   * Write a single block. If block with such hash exists, db will overwrite
+   * it.
+   * @param block to be written in a batch
+   * @return true if block already existed in db and we overwrote it. False
+   * otherwise.
+   */
+  virtual void put(const stored_block_t& block) = 0;
+
+  /**
+   * Remove a single block from storage identified by its hash.
+   * @param hash block hash
+   * @return true if removed, false if no such element found.
+   */
+  virtual bool removeByHash(const hash_t& hash) = 0;
+
+  /**
+   * Remove potentially many blocks at given height.
+   * @param height block height
+   * @return true if at least one block removed, false otherwise.
+   */
+  virtual int removeByHeight(height_t height) = 0;
```

Comment:
true if at least one block removed-> probably this was intended to return the number of blocks removed